container/factory.go

```diff
@@ -89,30 +86,27 @@ func HasFactories() bool {
 }
 
 // Create a new ContainerHandler for the specified container.
-func NewContainerHandler(name string, inHostNamespace bool) (ContainerHandler, bool, error) {
+func NewContainerHandler(name string, inHostNamespace bool) (ContainerHandler, error) {
 	factoriesLock.RLock()
 	defer factoriesLock.RUnlock()
 
 	// Create the ContainerHandler with the first factory that supports it.
 	for _, factory := range factories {
-		canHandle, canAccept, err := factory.CanHandleAndAccept(name)
+		handler, ignore, err := factory.NewContainerHandler(name, inHostNamespace)
```

Comment:
but doesn't this prevent NewContainerHandler from returning other errors that might occur?